src/compute/src/render/context.rs

```diff
@@ -1070,26 +1054,93 @@ where
     /// type descriptions.
     fn specialized_arrange(
         name: &String,
-        oks: Collection<S, (Row, Row), i64>,
-        _key_types: Vec<ColumnType>,
+        oks: Collection<S, Row, i64>,
+        key: &Vec<MirScalarExpr>,
+        thinning: &Vec<usize>,
+        key_types: Vec<ColumnType>,
         _val_types: Vec<ColumnType>,
-    ) -> SpecializedArrangement<S> {
-        // TODO(vmarcos) Specialize arrangements to types here!
-        let oks = oks.mz_arrange::<RowSpine<Row, Row, _, _>>(name);
-        let flavor = SpecializedArrangement::RowRow(oks);
-        flavor
+    ) -> (SpecializedArrangement<S>, Collection<S, DataflowError, i64>) {
+        if Bytes9::valid_schema(&key_types) {
+            // 9-byte key specialization.
+            let (oks, errs) = oks.map_fallible(
+                "[9-byte] FormArrangementKey",
```

Comment:
Can you check how our other operator names look like? We should try to remain somewhat consistent to make it easier to work with these names (e.g. for statistics collection) and I have the feeling that we usually append additional information rather than prepending it.

Reply:
I am not sure if there is a standard (and this might be good input to #21723!), but I just went ahead with post-fixing the `[9-byte]` tag for now.